### PR TITLE
Add new recovery tests for cache modes with lazy writes

### PR DESCRIPTION
**Pull request by katlapinka**

### CODE REVIEW

File: test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        compare_files(source_file, target_file)
+
+    with TestRun.step("Cleanup core device and remove test files."):
+        Dd() \
+            .input("/dev/zero") \
+            .output(core_device.system_path) \
+            .block_size(Size(1, Unit.Blocks4096)) \
+            .count(10) \
+            .oflag("direct") \
+            .run()
```
Comment (mmichal10) [inline test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py:106]: Why is this needed?

File: test/functional/tests/lazy_writes/recovery/test_recovery_unplug.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        compare_files(source_file, target_file)
+
+    with TestRun.step("Cleanup core device and remove test files."):
+        Dd() \
+            .input("/dev/zero") \
+            .output(core_device.system_path) \
+            .block_size(Size(1, Unit.Blocks4096)) \
+            .count(10) \
+            .oflag("direct") \
+            .run()
```
Comment (mmichal10) [inline test/functional/tests/lazy_writes/recovery/test_recovery_unplug.py:170]: why is this needed?

Comment (katlapinka) [reply test/functional/tests/lazy_writes/recovery/test_recovery_unplug.py:170]: At the beginning I had more than one iteration in this test so I had to clean core before next iteration, but I changed it to parametrize, so it can be removed now

File: test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py (hunk 1 above)
Comment (katlapinka) [reply test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py:106]: At the beginning I had more than one iteration in this test so I had to clean core before next iteration, but I changed it to parametrize, so it can be removed now

File: test/functional/tests/lazy_writes/recovery/test_recovery_all_options.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        description: Verify that unflushed data can be safely recovered after reset.
+        pass_criteria:
+          - CAS recovers successfully after reboot
+          - No data corruption
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        cache_disk.create_partitions([Size(200, Unit.MebiByte)])
+        core_disk.create_partitions([Size(2000, Unit.MebiByte)] * 2)
```
Comment (Ostrokrzew) [inline test/functional/tests/lazy_writes/recovery/test_recovery_all_options.py:49]: Where do You use the second partition?

File: test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+          data flushing on raw device.
+        pass_criteria:
+          - CAS recovers successfully after reboot
+          - No data corruption
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        cache_disk.create_partitions([Size(2, Unit.GibiByte)])
+        core_disk.create_partitions([Size(16, Unit.GibiByte)] * 2)
```
Comment (Ostrokrzew) [inline test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py:43]: Where do You use the second partition?

File: test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+          data flushing on filesystem.
+        pass_criteria:
+          - CAS recovers successfully after reboot
+          - No data corruption
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        cache_disk.create_partitions([Size(2, Unit.GibiByte)])
+        core_disk.create_partitions([Size(16, Unit.GibiByte)] * 2)
```
Comment (Ostrokrzew) [inline test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py:123]: Where do You use the second partition?

File: test/functional/tests/lazy_writes/recovery/test_recovery_all_options.py (hunk 3 above)
Comment (katlapinka) [reply test/functional/tests/lazy_writes/recovery/test_recovery_all_options.py:49]: second partition is created to make sure that after reboot CAS loads with use of correct partition - that's why I use symlinks, because sometimes after reboot device name is changed but symlink stays the same.

Comment (Ostrokrzew) [reply test/functional/tests/lazy_writes/recovery/test_recovery_all_options.py:49]: Thanks for explanation.

File: test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py (hunk 4 above)
Comment (katlapinka) [reply test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py:43]: as above

File: test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py (hunk 5 above)
Comment (katlapinka) [reply test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py:123]: as above
